spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/publisher/PubSubPublisherTemplate.java

```diff
@@ -84,16 +84,25 @@ public PubSubPublisherTemplate setMessageConverter(PubSubMessageConverter pubSub
 	 */
 	@Override
 	public <T> ListenableFuture<String> publish(String topic, T payload, Map<String, String> headers) {
+		Assert.hasText(topic, "The topic can't be null or empty.");
```

Comment:
I don't think we need this here since we are going to delegate to the ` publish(final String topic, PubsubMessage pubsubMessage)` anyway.

Reply:
Fixing.